src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BookingsReportGenerator.kt

```diff
@@ -19,10 +19,13 @@ class BookingsReportGenerator : ReportGenerator<BookingsReportDataAndPersonInfo,
         bookingId = booking.bookingId,
         referralId = booking.referralId,
         referralDate = booking.referralDate,
-        personName = personInfo.tryGetDetails { "${it.firstName} ${it.surname}".trim() },
-        pncNumber = personInfo.tryGetDetails { it.otherIds.pncNumber },
+        personName = personInfo.tryGetDetails {
+          val nameParts = listOf(it.name.forename) + it.name.middleNames + it.name.surname
+          nameParts.joinToString(" ")
+        },
+        pncNumber = null,
         gender = personInfo.tryGetDetails { it.gender },
-        ethnicity = personInfo.tryGetDetails { it.offenderProfile.ethnicity },
+        ethnicity = personInfo.tryGetDetails { it.profile?.ethnicity },
```

Comment:
Q; Looks like we've changed from `PersonInfoResult` to `PersonSummaryInfoResult` which then changes how we get ethnicity. [I don't think we did this before on attempt 1](https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1100/files#diff-f07bbcf9104e9ae593b098a2ae7dd2fad95e30e3ad8fd9e96175991e78ae1c21R29). Is PersonSummaryInfoResult a newer and perhaps efficient way of handling this too?

Reply:
This was originally done [in this commit](https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1100/commits/98f5b53b2ac11b86323d515892ff39f900ba88c5). The two types are unrelated beyond the similarities in their name, as `PersonInfoResult`s come from the Community API and `PersonSummaryInfoResult`s come from the Delius Integration API, and they return data of different shapes. The value of `PersonSummaryInfoResult` is that we can get many of them per request instead of just one but it has less information than `PersonInfoResult`, so I don't think it would be appropriate to replace `PersonInfoResult` with `PersonSummaryInfoResult` in every situation.